core/src/wsv.rs

```diff
@@ -279,23 +279,155 @@ impl WorldStateView {
     /// you likely have data corruption.
     /// - If trigger execution fails
     /// - If timestamp conversion to `u64` fails
-    #[iroha_futures::telemetry_future]
-    #[log(skip(self, block))]
     #[allow(clippy::expect_used)]
-    pub async fn apply(&self, block: VersionedCommittedBlock) -> Result<()> {
+    pub fn apply(&mut self, block: VersionedCommittedBlock) -> Result<()> {
```

Comment:
This function at the very least needs to be broken up, it's too long to be practical.

Reply:
Yes but not now. This function was created by unifying many 'inspect with closure' calls in order to satisfy the borrow checker. We will have to rewrite this when we move to a patch based way of doing things.